### PR TITLE
[UXE-3646] feat: create add payment method drawer

### PR DESCRIPTION
**Pull request by aloisio-m-bastian**
### What is the new behavior introduced by this PR?
Add the creation form to payment methods using the stripe-js library to integrate with the api.
image

### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does it have a link on Figma?

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [x] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: src/services/billing-services/create-payment-method-service.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  return `${apiKeyError}: ${apiValidationError}`
+}
+
+
+/**
+ * @param {Object} httpResponse - The HTTP response object.
+ * @param {Object} httpResponse.body - The response body.
+ * @param {String} httpResponse.statusCode - The HTTP status code.
+ * @returns {string} The result message based on the status code.
```
Comment (matheus-sunderhus) [inline src/services/billing-services/create-payment-method-service.js:42]: @returns {Object} The result object with feedback message

File: src/templates/add-payment-method-block/index.vue [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  import Sidebar from 'primevue/sidebar'
+  import { useAccountStore } from '@/stores/account'
+  import FeedbackFish from '@/templates/navbar-block/feedback-fish'
+  import InlineMessage from 'primevue/inlinemessage'
+  import FormHorizontal from '@/templates/create-form-block/form-horizontal'
+  import LabelBlock from '@/templates/label-block'
+  defineOptions({
+    name: 'add-payment-method-block'
+  })
+  const stripePromise = inject('stripe')
```
Comment (matheus-sunderhus) [inline src/templates/add-payment-method-block/index.vue:15]: mudar nome para stripePlugin.

Comment (matheus-sunderhus) [inline src/templates/add-payment-method-block/index.vue:15]: Existiria a possibilidade de injetarmos isso via props ao invés de usar o provide/inject?

File: src/templates/add-payment-method-block/index.vue [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+      required: true
+    }
+  })
+
+  const toast = useToast()
+  const showGoBack = ref(false)
+
+  onMounted(async () => {
+    stripe.value = await stripePromise
+    elements.value = stripe.value.elements()
```
Comment (matheus-sunderhus) [inline src/templates/add-payment-method-block/index.vue:38]: trocar nome de elements para stripeComponents

File: src/templates/add-payment-method-block/index.vue [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    createService: {
+      type: Function,
+      required: true
+    }
+  })
+
+  const toast = useToast()
+  const showGoBack = ref(false)
+
+  onMounted(async () => {
```
Comment (matheus-sunderhus) [inline src/templates/add-payment-method-block/index.vue:36]: refatorar função onMounted para chamar funções menores em sequência, evitando que o ciclo de vida fique com muita lógica, exemplo"
onMounted(async()=>{
loadStripeComponents() // parte do await e do stripe.value.elements()...
customizaStripeComponentsStyles() // parte do styles...
addToTemplateStripeComponents() // partes que está usando o .create() e mount()
})

**Review by matheus-sunderhus — COMMENTED** [review]
minor changes

File: src/templates/add-payment-method-block/index.vue (hunk 2 above)
Comment (aloisio-m-bastian) [reply src/templates/add-payment-method-block/index.vue:15]: posso colocar via prop e o inject uma camada acima, seria essa a sugestão?

File: src/services/billing-services/create-payment-method-service.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * @param {string} key - The error key of error schema.
+ * @returns {string} The result message based on the status code.
+ */
+const extractErrorKey = (errorSchema, key) => {
+  return `${errorSchema[key]}`
+}
+
+/**
+ * @param {Object} httpResponse - The HTTP response object.
+ * @param {Object} httpResponse.body - The response body.
```
Comment (vinigfer) [inline src/services/billing-services/create-payment-method-service.js:27]: We only have `httpResponse` as a parameter right?
```suggestion
```

File: src/services/billing-services/create-payment-method-service.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  const apiKeyError = Object.keys(httpResponse.body)[0]
+  const apiValidationError = extractErrorKey(httpResponse.body, apiKeyError)
+
+  return `${apiKeyError}: ${apiValidationError}`
+}
+
+/**
+ * @param {Object} httpResponse - The HTTP response object.
+ * @param {Object} httpResponse.body - The response body.
+ * @param {String} httpResponse.statusCode - The HTTP status code.
```
Comment (vinigfer) [inline src/services/billing-services/create-payment-method-service.js:40]: Similar to previous comment
```suggestion
```

File: src/templates/add-payment-method-block/index.vue [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+    position="right"
+    :pt="{
+      root: { class: 'max-w-4xl w-full' },
+      headercontent: { class: 'flex justify-content-between items-center w-full pr-2' },
+      content: { class: 'p-8' }
+    }"
+  >
+    <template #header>
+      <h2>Add Payment Method</h2>
+      <FeedbackFish />
```
Comment (vinigfer) [inline src/templates/add-payment-method-block/index.vue:156]: Does it make sense to add FeedbackFish? We are actually thinking about getting rid of it

File: src/templates/add-payment-method-block/index.vue [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+        @onSubmit="handleSubmit"
+        :inDrawer="true"
+        :loading="isSubmitting"
+      />
+    </div>
+  </Sidebar>
+</template>
+
+<style scoped>
+  /* Estilos para os elementos Stripe */
```
Comment (vinigfer) [inline src/templates/add-payment-method-block/index.vue:246]: English please

**Review by lucasmendes21 — DISMISSED** [review]

**Comment by matheus-sunderhus** [comment]
Adicionar Screenshot da implementação com nova feature

File: src/templates/add-payment-method-block/index.vue [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+        card_address_zip: accountData.postal_code,
+        card_country: accountData.country,
+        stripe_token: token.id,
+        card_id: token.card.id,
+        card_brand: token.card.brand,
+        card_holder: token.card.name,
+        card_last_4_digits: token.card.last4,
+        card_expiration_month: token.card.exp_month,
+        card_expiration_year: token.card.exp_year
+      }
```
Comment (matheus-sunderhus) [inline src/templates/add-payment-method-block/index.vue:125]: englobar no Try e adicionar um finally para colocar o valor de isSubmiting para false

File: src/templates/add-payment-method-block/index.vue [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+        emit('onSuccess', response)
+        showToast('success', response.feedback)
+        showGoBack.value = props.showBarGoBack
+        toggleDrawerVisibility(false)
+      } catch (error) {
+        emit('onError', error)
+        showToast('error', error)
+      }
+    }
+    isSubmitting.value = false
```
Comment (matheus-sunderhus) [inline src/templates/add-payment-method-block/index.vue:137]: mover essa linha para o finally mencionado acima.

**Review by matheus-sunderhus — COMMENTED** [review]
minor changes to go to Stage env

**Review by HerbertJulio — APPROVED** [review]

**Review by pauloSF0 — APPROVED** [review]

**Pull request merged**